qa/tasks/quiescer.py

```diff
@@ -0,0 +1,409 @@
+        stdout = None
+
+        while leader is None and not self.is_stopped:
+            leader = self.fs.get_var('qdb_leader')
+            if leader is None:
+                self.logger.warn("Couldn't get quiesce db leader from the mds map")
+                self.sleep_unless_stopped(5)
+
+        while leader is not None and not self.is_stopped:
+            command = ['tell', f"mds.{leader}", 'quiesce', 'db']
+            command.extend(args)
+            self.logger.debug("Running ceph command: '%s'" % " ".join(command))
+            result = self.fs.run_ceph_cmd(args=command, check_status=False, stdout=StringIO(),
+                                          # (quiesce_timeout + guard) is a sensible cmd timeout
+                                          # for both `--quiesce --await` and `--release --await`
+                                          # It is an overkill for a query,
+                                          # but since it's just a safety net, we use it unconditionally
+                                          timeoutcmd=self.quiesce_timeout+self.QDB_CMD_TIMEOUT_GUARD)
+            rc, stdout = result.exitstatus, result.stdout.getvalue()
+            if rc == errno.ENOTTY:
+                try:
+                    resp = json.loads(stdout)
+                    leader = int(resp['leader'])
+                    self.logger.info("Retrying a quiesce db command with leader %d" % leader)
+                except Exception as e:
+                    self.logger.error("Couldn't parse ENOTTY response from an mds with error: %s\n%s" % (str(e), stdout))
+                    self.sleep_unless_stopped(5)
+            else:
+                break
+
+        return (rc, stdout)
+
+    def dump_ops_all_ranks(self, dump_tag):
+        remote_paths = []
+
+        # begin by executing dump on all ranks
+        for info in self.fs.get_ranks():
+            name = info['name']
+            rank = info['rank']
+            remote_path = f"/tmp/mds.{name}-ops-{dump_tag}"
+            self.logger.debug(f"Dumping ops on rank {rank} ({name}) to a remote file {remote_path}")
+            try:
+                _ = self.fs.rank_tell(['ops', '--flags=locks', f'--path={remote_path}'], rank=rank)
+                remote_paths.append((info, remote_path))
+            except Exception as e:
+                self.logger.error(f"Couldn't execute ops dump on rank {rank}, error: {e}")
+
+        # now get the ops from the files
+        for info, remote_path in remote_paths:
+            try:
+                name = info['name']
+                rank = info['rank']
+                mds_remote = self.fs.mon_manager.find_remote('mds', name)
+                blob = misc.get_file(mds_remote, remote_path, sudo=True).decode('utf-8')
+                self.logger.debug(f"read {len(blob)}B of ops from '{remote_path}' on mds.{rank} ({name})")
+                ops_dump = json.loads(blob)
+                out_name = f"{self.ctx.archive}/ops-{dump_tag}-mds.{name}.json"
```

Comment:
I wanted to avoid that by creating an elaborate file naming. There are several jobs where it worked fine, like [here](http://qa-proxy.ceph.com/teuthology/leonidus-2024-04-23_17:44:29-fs-wip-lusov-quiescer-distro-default-smithi/7670097/), for example.

Reply:
I wanted to avoid having to create directories. The file name is already unique and hierarchical, so I'd like to leave it as is if it's OK with you

Reply:
It's okay for now. We can change it later if it clutters too much.